twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/WorkspaceChannelList.tsx

```diff
@@ -24,36 +37,31 @@ export default () => {
           suffix={
             <Icon type="search" className="m-icon-small" style={{ color: 'var(--grey-dark)' }} />
           }
-          autoFocus
           placeholder={Languages.t('scenes.client.channelbar.workspacechannellist.autocomplete')}
-          onChange={(e: React.ChangeEvent<HTMLInputElement>) => console.log(channels)}
+          value={search}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => setSearch(event.target.value)}
+          autoFocus
         />
       </Row>
-
       <PerfectScrollbar
         style={{ maxHeight: '250px' }}
         component="div"
         options={{ suppressScrollX: true, suppressScrollY: false }}
       >
-        {channels &&
-          channels.map((channel: ChannelResource, index: number) => {
+        {autoChannels
+          .filter(({ name }) => name.toUpperCase().indexOf(search.toUpperCase()) > -1)
+          .map((channel, index) => {
             return (
-              <div key={`${channel.data.id}_${index}`}>
+              <div key={`${channel.id}_${index}`}>
                 <WorkspaceChannelRow channel={channel} />
                 <Divider style={{ margin: 0 }} />
               </div>
             );
           })}
-        {channels.length === loadedChannels && (
-          <Row align="middle" justify="center" className="y-margin">
-            <Typography.Link
-              style={{ color: 'var(--grey-dark)' }}
-              onClick={() => setLoadedChannels(loadedChannels + 3)}
-            >
-              {Languages.t('scenes.client.channelbar.channelmemberslist.loader')}
-            </Typography.Link>
-          </Row>
-        )}
+        {!autoChannels.filter(({ name }) => name.toUpperCase().indexOf(search.toUpperCase()) > -1)
```

Comment:
Are you trying to capitalize first letter ? 

If yes, you can use `capitalize` from lodash.

https://lodash.com/docs/#capitalize

```js
import { capitalize } from 'lodash';
....

 {!autoChannels.filter(({ name }) => capitalize(name)
```

Reply:
No, the goal here is to search for string into a string, regardless of case. So it is working.
@RkAiman can you implement something to also ignore accents in your next PR for members search ? (I send you a DM too)